File: Project_Python3/StackSpeedTest/StackSpeedTest.py
# -*- coding: utf-8 -*-

import time

def main(max):
    time0 = time.time()

    stack = []

    for i in range(max):
        stack.append(i)

    print("stack.push({0:d}) done.".format(max))
    print("stack.index(0) ... {0:d}".format(stack.index(0)))

    while len(stack) > 0:
        """
        size = len(stack)
        peek = stack[-1]
        pop = stack.pop()
        print("size = {0:d}, peek = {1:d}, pop = {2:d}".format(size, peek, pop))
        """
        stack.pop()

    print("stack.pop({0:d}) done.".format(0))
    print("len(stack) == {0:d}".format(len(stack)))

    time1 = time.time()

    print("Execute time : {0:f} [ms]\n".format((time1 - time0)*1000))

File: Project_Java_Stack/StackSpeedTest.java
import java.util.*;

public class StackSpeedTest {
    public void Main(int max) {
        long start = System.currentTimeMillis();

        Stack<Integer> stack = new Stack<>();

        for (int i = 0; i < max; i++) {
            stack.push(i);
        }

        System.out.println(String.format("stack.push(%d) done.", max));
        System.out.println(String.format("stack.Search(0) ... %d", stack.search(0)));

        while (stack.empty() == false) {
            /*
            int size = stack.size();
            int peek = stack.peek();
            int pop = stack.pop();
            System.out.println(formatter.format("size = %d, peek = %d, pop = %d\n", size, peek, pop));
            */
            stack.pop();
        }

        System.out.println(String.format("stack.pop(%d) done.", 0));
        System.out.println(String.format("stack.empty()   ... %s", Boolean.toString(stack.empty())));

        long end = System.currentTimeMillis();

        System.out.println("Execute time : " + (end - start)  + " [ms]\n");
    }
}
